devenv/pythons.py

```diff
@@ -1,17 +1,51 @@
 from __future__ import annotations
 
 import os
+import shutil
+import subprocess
 
+from devenv.constants import CI
 from devenv.constants import root
 from devenv.lib import archive
 
 
+def _check_version(exe: str, version: str) -> bool:
+    """Returns true if the executable at the given path is a python interpreter
+    with a semver-compatible version."""
+
+    try:
+        line = (
+            subprocess.check_output([exe, "--version"], text=True)
```

Comment:
much more reliable to `sys.executable -c 'import json, sys; print(json.dumps(sys.version_info[:3]))'